initializer/mount.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package main
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+
+	"github.com/edgelesssys/contrast/internal/logger"
+	"github.com/edgelesssys/contrast/internal/mount"
+	"github.com/spf13/cobra"
+)
+
+// luksVolume struct holds the representative attributes related to a LUKS encrypted volume.
+type luksVolume struct {
+	devicePath           string
+	mappingName          string
+	volumeMountPoint     string
+	encryptionPassphrase string
+}
+
+func must(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// NewSetupEncryptedMountCmd creates a Cobra subcommand of the initializer to set up the specified encrypted LUKS volume.
+func NewSetupEncryptedMountCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "setupEncryptedMount -d [device-path] -m [mount-point]",
+		Short: "setupEncryptedMount on block device at [device-path] with decrypted mapper device at [mount-point]",
+		Long: `Set up an LUKS encrypted VolumeMount on the provided VolumeDevice
+		located at the specified [device-path] and mount the decrypted mapper
+		device to the provided [mount-point].
+
+		In certain deployments, we require a persistent volume claim configured
+		as block storage to be encrypted by the initializer binary.
+		Therefore we expose the defined PVC as a block VolumeDevice to our
+		initializer container. This allows the initializer to setup the
+		encryption on the block device located at [device-path] using cryptsetup,
+		the encryption passphrase is derived from the UUID of the LUKS formatted
+		block device and the current workload secret.
+
+		The mapped decrypted block device can then be shared with other containers
+		on the pod by setting up a shared VolumeMount on the specified [mount-point],
+		where the mapper device will be mounted to.`,
+		RunE: setupEncryptedMount,
+	}
+	cmd.Flags().StringP("device-path", "d", "/dev/csi0", "path to the volume device to be encrypted")
+	cmd.Flags().StringP("mount-point", "m", "/state", "mount point of decrypted mapper device")
+	must(cmd.MarkFlagRequired("device-path"))
+	must(cmd.MarkFlagRequired("mount-point"))
+
+	return cmd
+}
+
+// parseSetupEncryptedMountFlags returns a luksVolume, representing the provided flags to the subcommand, which is then used to setup the encrypted volume mount.
+func parseSetupEncryptedMountFlags(cmd *cobra.Command) (*luksVolume, error) {
+	devicePath, err := cmd.Flags().GetString("device-path")
+	if err != nil {
+		return nil, err
+	}
+	mountPoint, err := cmd.Flags().GetString("mount-point")
+	if err != nil {
+		return nil, err
+	}
+	hash := md5.Sum([]byte(devicePath + mountPoint))
```

Comment:
This is fine, but I already see code scanners etc. complaining about the use of md5. Any objections to just using sha256?